pkg/api/types_test.go

```diff
@@ -392,7 +392,48 @@ func TestTotalNodes(t *testing.T) {
 		}
 	}
 }
+func TestMasterAvailabilityProfile(t *testing.T) {
+	cases := []struct {
+		p              Properties
+		expectedISVMSS bool
```

Comment:
I would just change this to one variable. It's either AvailablitySet or VirtualMachineScaleSet anyway once it foes past validation.